File: Constructor Overloading .cpp
#include <iostream>
using namespace std;

class Person {
private:
    string name;
    int age;

public:
    // Constructor 1
    Person(string n, int a) : name(n), age(a) {}

    // Constructor 2
    Person(string n) : name(n), age(0) {}

    void show() {
        if (age == 0)
            cout << "Name: " << name << endl;
        else
            cout << "Name: " << name << ", Age: " << age << endl;
    }
};

int main() {
    Person person1("John", 30);
    Person person2("Alice");
    person1.show();
    person2.show();
    return 0;
}

File: Constructor Overloading .py
# 3. Constructor Overloading
# Constructor Overloading allows a class to have more than one constructor, each with different parameters. This lets you create objects in multiple ways.\


class Person:
    def __init__(self, name=None, age=None):
        self.name = name
        self.age = age

    def show(self):
        if self.name and self.age:
            return f"Name: {self.name}, Age: {self.age}"
        elif self.name:
            return f"Name: {self.name}"
        else:
            return "No information available"

person1 = Person("John", 30)
person2 = Person("Alice")
print(person1.show())
print(person2.show())
